controllers/session_test.go

```diff
@@ -0,0 +1,102 @@
+package controllers_test
+
+import (
+	"net/http"
+	"net/url"
+	"strings"
+
+	_ "google-scraper/initializers"
+	. "google-scraper/tests/fabricators"
+	. "google-scraper/tests/testing_helpers"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("SessionController", func() {
+	Describe("GET /login", func() {
+		Context("given the user is a guest user", func() {
+			It("returns 200 status code", func() {
+				response := MakeRequest("GET", "/login", nil)
+
+				Expect(response.StatusCode).To(Equal(http.StatusOK))
+			})
+		})
```

Comment:
Shouldn't there be a case when the user is already authenticated too? The same applies to the tests for RegistrationController.

Another option would be to add test for `base.go` by creating a fake controller and adding the base controller to it. Then the tests for requiring guest and authenticated users could be done once only.